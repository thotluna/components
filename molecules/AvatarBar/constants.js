export const options = {
  sizes: ['sm', 'lg'],
}
