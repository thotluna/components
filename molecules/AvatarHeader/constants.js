export const options = {
  size: ['sm', 'md'],
}
