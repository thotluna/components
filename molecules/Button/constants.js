export const options = {
  colors: ['base', 'primary', 'secondary'],
}
