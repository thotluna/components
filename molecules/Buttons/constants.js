export const options = {
  size: ['full', 'short', 'mini'],
  color: ['primary', 'secondary', 'inverted'],
}
