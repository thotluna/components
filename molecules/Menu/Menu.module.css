.menu {
  position: relative;
}

.menu-list-items {
  position: absolute;
  z-index: 9;
  top: 48px;
  min-width: 100px;
  padding: 0.5em;
  background-color: var(--color-brand-mist-blue);
  font-size: var(--font-size-md);
}

.menu-list-items.position-left {
  left: 0;
}

.menu-list-items.position-right {
  right: 0;
}

.menu-item a {
  color: var(--color-font-base);
  text-decoration: none;
}

.menu-item a:hover {
  text-decoration: underline;
}
