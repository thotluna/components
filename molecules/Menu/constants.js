export const options = {
  position: ['left', 'right'],
}
