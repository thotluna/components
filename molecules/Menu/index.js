export { default } from './Menu'
export { options } from './constants'
export { default as styles } from './Menu.module.css'
