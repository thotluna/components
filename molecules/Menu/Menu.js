import React, { useState } from 'react'
import PropTypes from 'prop-types'
import styles from './Menu.module.css'
import { options } from './constants'
import { getClasses } from '../../helpers/styles'

import Button from '../Button'

export const Menu = ({ listLink, active, position }) => {
  const [show, setShow] = useState(active)
  const getStyles = getClasses(styles)({ position })
  return (
    <nav className={getStyles('menu')}>
      <Button
        ariaLabel="menu"
        icon="menu"
        id="menu"
        handelOnClick={() => {
          setShow(!show)
        }}
      />
      {show && (
        <div className={getStyles('menu-list-items', ['position'])}>
          {listLink.length &&
            listLink.map((item, index) => (
              <div key={index} className={getStyles('menu-item')}>
                <a
                  onClick={() => setShow(!show)}
                  aria-current={item.name}
                  href={item.link}
                >
                  {item.name}
                </a>
              </div>
            ))}
        </div>
      )}
    </nav>
  )
}

Menu.propTypes = {
  listLink: PropTypes.arrayOf(
    PropTypes.shape({
      link: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    })
  ),
  position: PropTypes.oneOf(options.position),
  active: PropTypes.bool,
}

Menu.defaultProps = {
  position: 'left',
  active: false,
}

export default Menu
