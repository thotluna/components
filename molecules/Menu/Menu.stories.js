import React from 'react'
import Menu from '.'
import { getOptionsArgTypes, getVariusTemplate } from '../../helpers/storybook'
import { options } from './constants'

const Template = (args) => <Menu {...args} />
const ListTemplae = getVariusTemplate(Menu)

export default {
  title: 'Molecules/Menu',
  menu: Menu,
  args: {
    listLink: [
      { link: '#', name: 'Home' },
      { link: '#', name: 'Task' },
    ],
  },
  argTypes: {
    position: getOptionsArgTypes(options.position),
  },
  parameters: {
    __sb: { mh: '600', fd: 'row', jc: 'space-between' },
  },
}

export const Default = Template.bind({})

export const Active = Template.bind({})
Active.args = {
  active: true,
}

export const Posicion = ListTemplae.bind({})
Posicion.args = {
  items: options.position.map((position) => ({ position })),
}
