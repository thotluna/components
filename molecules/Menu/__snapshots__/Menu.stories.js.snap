// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/Menu Active 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "space-between",
      "maxHeight": "600",
      "padding": "2em",
    }
  }
>
  <nav
    className="menu"
  >
    <button
      aria-label="menu"
      className="button color-base is-round"
      id="menu"
      onClick={[Function]}
    >
      <span
        className="icon color-base size-sm"
      >
        <svg
          fill="currentColor"
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
            transform="translate(2 2)"
          >
            <path
              d="M8.5 9.5c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1zm-4 0c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1zm8 0c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1z"
            />
          </g>
        </svg>
      </span>
    </button>
    <div
      className="menu-list-items position-left"
    >
      <div
        className="menu-item"
      >
        <a
          aria-current="Home"
          href="#"
          onClick={[Function]}
        >
          Home
        </a>
      </div>
      <div
        className="menu-item"
      >
        <a
          aria-current="Task"
          href="#"
          onClick={[Function]}
        >
          Task
        </a>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Storyshots Molecules/Menu Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "space-between",
      "maxHeight": "600",
      "padding": "2em",
    }
  }
>
  <nav
    className="menu"
  >
    <button
      aria-label="menu"
      className="button color-base is-round"
      id="menu"
      onClick={[Function]}
    >
      <span
        className="icon color-base size-sm"
      >
        <svg
          fill="currentColor"
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
            transform="translate(2 2)"
          >
            <path
              d="M8.5 9.5c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1zm-4 0c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1zm8 0c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1z"
            />
          </g>
        </svg>
      </span>
    </button>
  </nav>
</div>
`;

exports[`Storyshots Molecules/Menu Posicion 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "space-between",
      "maxHeight": "600",
      "padding": "2em",
    }
  }
>
  <nav
    className="menu"
  >
    <button
      aria-label="menu"
      className="button color-base is-round"
      id="menu"
      onClick={[Function]}
    >
      <span
        className="icon color-base size-sm"
      >
        <svg
          fill="currentColor"
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
            transform="translate(2 2)"
          >
            <path
              d="M8.5 9.5c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1zm-4 0c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1zm8 0c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1z"
            />
          </g>
        </svg>
      </span>
    </button>
  </nav>
  <nav
    className="menu"
  >
    <button
      aria-label="menu"
      className="button color-base is-round"
      id="menu"
      onClick={[Function]}
    >
      <span
        className="icon color-base size-sm"
      >
        <svg
          fill="currentColor"
          height={21}
          viewBox="0 0 21 21"
          width={21}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
            transform="translate(2 2)"
          >
            <path
              d="M8.5 9.5c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1zm-4 0c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1zm8 0c.5 0 1-.5 1-1s-.5-1-1-1-.999.5-.999 1 .499 1 .999 1z"
            />
          </g>
        </svg>
      </span>
    </button>
  </nav>
</div>
`;
